Tool Shed 2.0 fixes

nsoranzo opened the pull request:
- toolshed GitHub workflow fixes:
  - Remove duplicated cache key that prevented the tests from being run
  - Expand matrix to test all combinations on Python 3.7 and 3.11
  - Install node from setup-node action
- Fix issues reported by ``yarn build``
- Explicitly add playwright to dev dependencies
- Restore `GET /api/categories/<encoded_id>` and the `deleted` field in the response.
- Add some type annotations.
- Remove unused functions and re-exports.
- Misc bug fixes.

A bunch of ToolShed tests are failing, but they were simply not being run previously.
We can either merge this and fix the failing tests in separate PRs, or use this PR to push the fixes, but I don't have time to look into them currently.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

nsoranzo committed:
Fix duplicated cache key

nsoranzo committed:
Expand matrix to test all combinations on Python 3.7 and 3.11

nsoranzo committed:
Expand ``run:`` values to multiple lines

nsoranzo committed:
Fix issues reported by ``yarn build``

Introduced in https://github.com/galaxyproject/galaxy/pull/16787 .

nsoranzo committed:
Explicitly add playwright to dev dependencies

jmchilton commented on .github/workflows/toolshed.yaml:
```diff
@@ -19,24 +19,11 @@ jobs:
-        - test-install-client: 'standalone'
-          python-version: '3.10'
-          shed-api: 'v2'
-          shed-browser: 'playwright'
```
I was "attempting" to only testing a subset of configurations because I thought all against all on the these combinations would be too much testing for such a niche feature. Does this configuration double the number of tool shed tests we're running?

nsoranzo replied:
Yes, from 4 to 8. If you think that some of these combinations are not worth to be run on both Python versions, I can redo the matrix to run them only on Python 3.7.

nsoranzo committed:
Check that TOOL_SHED_TEST_BROWSER is a valid value

nsoranzo committed:
Make owner and name parameters of ``get_or_create_repository()`` not optional

nsoranzo committed:
Restore GET /api/categories/<encoded_id> in ToolShed 2.0

and the deleted field in the response.

Rationale: it is not particularly useful, but both:
GET /api/categories
GET /api/categories/<encoded_id>/repositories
work on TS 2.0, so it's weird that this one is missing.

Also, it was breaking BioBlend's
https://bioblend.readthedocs.io/en/latest/api_docs/toolshed/all.html#bioblend.toolshed.categories.ToolShedCategoryClient.show_category

nsoranzo committed:
Fix imports

Broken in commit 3238b212805f35aa17906ee51296a08b491e0e4c .

nsoranzo committed:
Install node from setup-node action

nsoranzo committed:
Fix `get_users_with_repo_alert()` call

Introduced in commit 1711e96dacf02724ae7d7f73d8cbc02201e88477 .

nsoranzo committed:
Add type annotations. Drop unused functions and re-exports.

nsoranzo committed:
Fix `index_repositories()` method

Broken in commit 77df9c9a73e6efd5233ec881a555855e913b3aaf .

jdavcs reviewed (approved):
Looks good to me!

nsoranzo committed:
Fix typo in query

Introduced in commit 3238b212805f35aa17906ee51296a08b491e0e4c .

nsoranzo committed:
Fix another typo

Introduced in commit 70c7f03be9ef02bcdd405820ad68f1772f43a25d .

nsoranzo committed:
Fix another typo

Introduced in commit 68831ddcfc661ca13ce7b26cc7383589f185c65a .

nsoranzo committed:
Type annotation fixes

nsoranzo committed:
Remove unused test config option

nsoranzo committed:
Configure temporary whoosh_index_dir for tests

Otherwise a `<GALAXY_ROOT>/database/toolshed_whoosh_indexes` would
be reused, leading to failing tests when running tests locally.

mvdbeek reviewed (approved):
Awesome, thanks a lot @nsoranzo!
